testing/tests/DevExpress.ui.widgets/treeViewParts/virtualMode.js

```diff
@@ -199,6 +199,39 @@ QUnit.test('Remove toggle icon after expand childless item', function(assert) {
     assert.equal($icons.length, 2);
 });
 
+QUnit.test('No custom expander icons should be visible after expand childless item', function(assert) {
+    const treeView = new TreeView(this.$element, {
+        dataSource: $.extend(true, [], data2),
+        dataStructure: 'plain',
+        expandButtonIcon: 'add',
+        collapseButtonIcon: 'minus',
+        virtualModeEnabled: true
+    });
+
+    const $expandIcons = this.$element.find(`.${internals.CUSTOM_EXPAND_ICON_CLASS}`);
+    assert.ok($expandIcons.eq(0).is(':visible'));
+    assert.ok($expandIcons.eq(1).is(':visible'));
+    assert.ok($expandIcons.eq(2).is(':visible'));
+
+    const $collapseIcons = this.$element.find(`.${internals.CUSTOM_COLLAPSE_ICON_CLASS}`);
+    assert.notOk($collapseIcons.eq(0).is(':visible'));
+    assert.notOk($collapseIcons.eq(1).is(':visible'));
+    assert.notOk($collapseIcons.eq(2).is(':visible'));
+
+    treeView.expandItem(16);
+
+    const $expandIconsAfterExpandChildlessNode = this.$element.find(`.${internals.CUSTOM_EXPAND_ICON_CLASS}`);
+
+    assert.ok($expandIconsAfterExpandChildlessNode.eq(0).is(':visible'));
+    assert.ok($expandIconsAfterExpandChildlessNode.eq(1).is(':visible'));
+    assert.notOk($expandIconsAfterExpandChildlessNode.eq(2).is(':visible'));
+
+    const $collapseIconsAfterExpandChildlessNode = this.$element.find(`.${internals.CUSTOM_COLLAPSE_ICON_CLASS}`);
+    assert.notOk($collapseIconsAfterExpandChildlessNode.eq(0).is(':visible'));
+    assert.notOk($collapseIconsAfterExpandChildlessNode.eq(1).is(':visible'));
+    assert.notOk($collapseIconsAfterExpandChildlessNode.eq(2).is(':visible'));
+});
```

Comment:
Cannot we simplify this test?
Seems like it tests several things at one time. It's not an easy task for me to understand what is going here

Moreover, it's not clear for me how we can "expand childness node")

Reply:
Ok i'll try)
Moreover, it's not clear for me how we can "expand childness node") - this is a virtual mode, so in some cases we dont know if parent contains childrem. we have similar test for default icons already